pkg/component/component.go

```diff
@@ -970,6 +981,18 @@ func GetComponentFromConfig(localConfig *config.LocalConfigInfo) (Component, err
 	return Component{}, nil
 }
 
+// This function returns a mapping of port and protocol.
+// So for a value of ports {"8080/TCP", "45/UDP"} it will return a map {"8080":
+// "TCP", "45": "UDP"}
+func getPortsProtocolMapping(ports []string) map[string]string {
+	oldPortsProtocol := make(map[string]string, len(ports))
+	for _, port := range ports {
+		portProtocol := strings.Split(port, "/")
+		oldPortsProtocol[portProtocol[0]] = portProtocol[1]
```

Comment:
this will panic if the mappings are incorrect
```
[mrinaldas@localhost nodejs-ex]$ odo config set ports 909,808 -f
 ✓  Local config successfully updated

Run `odo push --config` command to apply changes to the cluster
[mrinaldas@localhost nodejs-ex]$ odo describe 
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/openshift/odo/pkg/component.getPortsProtocolMapping(0xc0006136c0, 0x2, 0x2, 0x0)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/component/component.go:991 +0x177
github.com/openshift/odo/pkg/component.GetComponentFromConfig(0xc000ba7500, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/component/component.go:962 +0x6d6
github.
```

Reply:
There never was a check for this in the first place. IMO, we should add it in the documentation of `odo config set -h` so that users don't hit this.

Reply:
Yeah we need to document this and I think should add a check to config set ports too.

Reply:
> Yeah we need to document this

Done. I'll push the change. But here's how it looks:
```sh
Examples:

  # Set a configuration value in the local config
  odo config set Type java
  odo config set Name test
  odo config set MinMemory 50M
  odo config set MaxMemory 500M
  odo config set Memory 250M
  odo config set DebugPort 4040
  odo config set Ignore false
  odo config set MinCPU 0.5
  odo config set MaxCPU 2
  odo config set CPU 1
  odo config set Ports 8080/TCP,8443/TCP    <<------- this
  
  # Set a env variable in the local config
  odo config set --env KAFKA_HOST=kafka --env KAFKA_PORT=6639
```

> I think should add a check to config set ports too.

I'd prefer a separate issue to work this out.

Reply:
> I'd prefer a separate issue to work this out.

Cool :+1: Please open a issue so that we don't forget about it

Reply:
https://github.com/openshift/odo/issues/3036